### ISSUE TITLE
MissingBreakInSwitchStatementRule false positive

### ISSUE THREAD

**Issue by maplesteve** [oclint/oclint#16]
Hi @lqi,

me again ;-)

oclint reports violations for the following code:

```
switch (type) {
    case WMSDocumentTypeCategory:
        fileName = WMSFileNameCategoryDocument;
        break;

    case WMSDocumentTypeItem:
        fileName = WMSFileNameItemDocument;
        break;

    case WMSDocumentTypePlace:
        fileName = WMSFileNamePlaceDocument;
        break;

    default:
        break;
}
```

The violations are reported for every switch that tests on variables, that are defined as enums.

I wrote a test case to examine the behavior:

```
TEST(MissingBreakInSwitchStatementRuleTest, CasesHaveBreaks2)
{
    testRuleOnCode(new MissingBreakInSwitchStatementRule(), "void aMethod(int a) { \n\
    typedef enum WMSDocumentType { \n\
    WMSDocumentTypeCategory  = 1 << 0, \n\
WMSDocumentTypeItem = 1 << 1, \n\
WMSDocumentTypePlace = 1 << 2 \n\
} WMSDocumentType; \n\
switch(a){\n\
case WMSDocumentTypeCategory:     \n\
\tbreak;    \n\
case WMSDocumentTypeItem:     \n\
\tbreak;    \n\
case WMSDocumentTypePlace:     \n\
\tbreak;    \n\
} }");
}
```

This test _does not fail_ - so I assume, that oclint get's confused by the usage of enums somehow. They are declared in the .pch file. May that be the issue?

I assume, that the same behaviour applies to at least also the SwitchStatementsShouldHaveDefaultRule.

What do you think?
